auth/api/auth/v2/session.go

```diff
@@ -0,0 +1,46 @@
+package v2
+
+import (
+	"github.com/google/uuid"
+	"net/url"
+	"sync"
+)
+
+type SessionManager struct {
+	sessions *sync.Map
+}
+
+func (s *SessionManager) Create(session Session) string {
+	// TODO: Session expiration
+	// TODO: Session storage
+	// TODO: Session pinning and other safety measures (see OAuth2 Threat Model)
+	id := uuid.NewString()
+	s.sessions.Store(id, session)
+	return id
+}
+
+func (s *SessionManager) Get(id string) *Session {
+	session, ok := s.sessions.Load(id)
+	if !ok {
+		return nil
+	}
+	result := session.(Session)
+	return &result
+}
+
+type Session struct {
```

Comment:
comment: [RFC6749 3.1 Authorization Endpoint](https://datatracker.ietf.org/doc/html/rfc6749#section-3.1)
>   Parameters sent without a value MUST be treated as if they were
   omitted from the request.  The authorization server MUST ignore
   unrecognized request parameters.  Request and response parameters
   MUST NOT be included more than once.

Meaning `param == ""` can be interpreted as the parameter is not present in the authorization request

(just nothing this as confirmation that "" == nil in this case)

Reply:
exactly, `"" != nil` but we can interpret it as such  instead of using annoying  `*string`